src/api/router.go

```diff
@@ -126,6 +85,28 @@ func (api *Api) ApiRouter() *gin.Engine {
 		v1.GET("/stacks/:namespace/services/:service_id/cd_url", api.ServiceCDAddr)
 	}
 
+	for _, plugin := range apiplugin.ApiPlugins {
+		if plugin.Instance != nil {
+			switch plugin.Name {
+			case apiplugin.Search:
+				searchApi, ok := plugin.Instance.(*search.SearchApi)
```

Comment:
How about add this section as a callback of apiplugin Init?

such as apiplugin.InitCallback = func() {
...
}

then call a apiplugin.InitCallback here